Assets/Scripts/Player/TempGoal.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TempGoal : MonoBehaviour
+{
+    private bool hasReachedGoal = false;
+
+    private void OnTriggerEnter2D(Collider2D other)
+    {
+        if (!hasReachedGoal && other.gameObject.CompareTag("Finish"))
+        {
+            hasReachedGoal = true; // Set the flag so this code won't run again
+            Debug.Log("Goal reached!");
+            GameEventsManager.current.GoalReached();
+
+            // Teleport the player back to the starting point
+            transform.position = new Vector3(5.72f, -3.001f, 0);
```

Comment:
Same here, the values should be saved from somewhere and loaded

Reply:
By saving, do you mean at the top of the class, or is there a neater way to do this? As in, can we save all these values in a config file and access them all from there?

Reply:
We can save in a config too. For here tho I was thinking on Start() we can just save our current position and then use that here instead of having to have numbers in the code

Reply:
Makes total sense, thanks!